b9lab-content/3-main-concepts/02-architecture.md

```diff
@@ -1 +1,294 @@
+order: 2
+description: Architecture of a blockchain application built with Cosmos.
+---
+
 # Blockchain App Architecture
+
+## What is Tendermint
+
+Created in 2014, Tendermint accelerates the development of distinct blockchains by attending to general concerns so that those concerns don’t need to be recreated by developers in each case. You may already be using Tendermint without awareness of this since blockchains including [Hyperledger Burrow](https://hyperledger.github.io/burrow/#/) and [Binance Chain](https://www.binance.org/en/smartChain)  use Tendermint.
+
+More specifically, Tendermint modules attend to consensus and networking, two important components of any blockchain. This frees developers to consider what the blockchain should do at an application level without descending into lower-level blockchain concerns such as peer discovery, block propagation, consensus and transaction finalization. Without Tendermint, developers would be forced to build software to address those concerns which would add additional time, complexity and cost to the blockchain applications they have in mind.
+
+![blockchain application architecture overview](./images/architecture_overview.png)
+
+A blockchain node for an application-focused Cosmos blockchain consists of a state-machine built with the Cosmos SDK and consensus and networking which is handled by Tendermint Core.
+
+As you might expect, Tendermint itself is modular and flexible. We will concentrate on the Tendermint Core used in Cosmos.
+
+## Consensus in Tendermint Core and Cosmos
+
+In summary, Tendermint Core is a high-performance, consistent, and secure **consensus** module with strict fork accountability. It relies on Proof-of-Stake (PoS) with delegation and [Practical Byzantine Fault Tolerance](https://github.com/tendermint/tendermint). Participants signal support for well-behaved, reliable nodes that create and confirm blocks. Users signal support by staking ATOM with the possibility of acquiring a share of network transaction fees but also sharing in the risk of reduced returns or even losses should the node become unreliable.
+
+Network participants are incentivized to stake ATOM in the fittest nodes that they deem most likely to provide a dependable service and to withdraw their support should conditions change. In this way, a Cosmos blockchain is expected to adjust the validator configuration and continue even in adverse conditions.
+
+In more detail, only the top 100 nodes by staked ATOM, the **validators**, participate in the transaction finalization process. The privilege of creating a block is awarded in proportion to the ATOM tokens committed. For example, if a given validator has 15% of the ATOM staked to the top 100 nodes, then it can expect to receive the block creation privilege 15% of the time. The block is broadcasted to the other validators, who are expected to respond promptly and correctly. They will absorb penalties for failing to do so.
```

Comment:
```suggestion
In more detail, only the top 150 nodes by staked ATOM, the **validators**, participate in the transaction finalization process. The privilege of creating a block is awarded in proportion to the voting power a validator has. Voting power is calculated as all ATOM tokens staked by a validator and its delegates. For example, if a given validator's voting power would be 15% of the total voting power of all validators, then it can expect to receive the block creation privilege 15% of the time. The block is broadcasted to the other validators, who are expected to respond promptly and correctly. They will absorb penalties for failing to do so.
```
150 Nodes since this week.

Also, the 15% example isn't very clear to me. Do you mean 15% of all staked ATOM? As far as I'm aware, a validator does not stake his ATOM to the top 100 nodes, it just stakes them. Feels slightly confusing. Plus we also have to consider delegate stakes to calculate total voting power.